core/dbt/parser/schemas.py

```diff
@@ -488,11 +488,28 @@ def render_test_update(self, node, config, builder):
         if (macro_unique_id in
                 ['macro.dbt.test_not_null', 'macro.dbt.test_unique']):
             self.update_parsed_node(node, config)
+            # manually set configs
             if builder.severity() is not None:
                 node.unrendered_config['severity'] = builder.severity()
                 node.config['severity'] = builder.severity()
             if builder.enabled() is not None:
+                node.unrendered_config['enabled'] = builder.enabled()
                 node.config['enabled'] = builder.enabled()
+            if builder.where() is not None:
+                node.unrendered_config['where'] = builder.where()
+                node.config['where'] = builder.where()
+            if builder.limit() is not None:
+                node.unrendered_config['limit'] = builder.limit()
+                node.config['limit'] = builder.limit()
+            if builder.warn_if() is not None:
+                node.unrendered_config['warn_if'] = builder.warn_if()
+                node.config['warn_if'] = builder.warn_if()
+            if builder.error_if() is not None:
+                node.unrendered_config['error_if'] = builder.error_if()
+                node.config['error_if'] = builder.error_if()
+            if builder.fail_calc() is not None:
+                node.unrendered_config['fail_calc'] = builder.fail_calc()
+                node.config['fail_calc'] = builder.fail_calc()
```

Comment:
I'd prefer to see this in the builder as a method that just returns the configs. But we can leave that for later.